src/cargo/util/command_prelude.rs

```diff
@@ -119,7 +119,7 @@ pub trait CommandExt: Sized {
         actual: &'static str,
     ) -> Self {
         let msg = format!(
-            "There is no `--{want}` for `cargo {command}`. Only `--{actual}` is supported."
+            "there is no `--{want}` for `cargo {command}`. Only `--{actual}` is supported."
```

Comment:
nit: should that be a `,` or `;` before `Only`?

Overall, I'd probably simplify it to `cargo {command} supports `--{actual}` instead` since the main message already has "unexpected argument".

Reply:
Updated! Thanks!

Reply:
I'm sorry, there is way to much back and forth on this.

Reading back over this, I realized what was missing or isn't clear.

```
`{want}` is the default for `cargo {command}`; instead `--actual` is supported
```

The downside to this is that it is failure specialized.  If we go with this wording, we should either
- Not use the helper
- Clarify the naming of the helper that it is about being the default mode
- Leave the message as is and go ahead and merge this

Reply:
> `{want}` is the default for `cargo {command}`; instead `--actual` is supported

I don't really get this. Do you mean `{actual}` is the default for `cargo {command}`; instead `--actual` is supported `?



> * Not use the helper
> * Clarify the naming of the helper that it is about being the default mode

The current helper name is `arg_unsupported_mode`. It helps to catch the unsupported mode instead of indicating the default mode. So I guess its name is OK?

Reply:
```
error: unexpected argument '--release' found

  tip: `cargo install` supports `--debug` instead.

Usage: cargo[EXE] install [OPTIONS] [crate]...

For more information, try '--help'.
```
Here, `--release` is unexpected because the `release` profile is the default.  `--debug` isn't really an alternative to use instead but is instead what we expose for controlling the profile.

So this isn't about an unsupported mode in this case but that the flag is unexpected because its redundant.

Maybe in other cases, a `arg_unsupported_mode` function with a message like that will make sense but I feel like here we should be focusing on the fact that the flag is redundant.

Reply:
Oh, I got it. Make sense. You are right it is a bad name and the tips are confusing.

Reply:
Updated. Thanks for your help!